File: Srims.Server.Business/Patents/PatentType.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Srims.Server.Business.Patents
{
    /// <summary>
    /// 专利类型
    /// </summary>
    public enum PatentType
    {
        /// <summary>
        /// 未知类型
        /// </summary>
        Unknown = 0,
        /// <summary>
        /// 发明专利
        /// </summary>
        Invention = 1,
        /// <summary>
        /// 实用新型
        /// </summary>
        Application = 2,
        /// <summary>
        /// 外观设计
        /// </summary>
        Design = 3,

    }
}

File: Srims.WebSite/javascript/srims/Patents/PatentType.js

if (!Srims.patents)
    Ext.namespace("Srims.patents");

Ext.namespace('Srims.patents.PatentType');

Srims.patents.PatentType.Unknown = 'Unknown';
Srims.patents.PatentType.Invention = 'Invention';
Srims.patents.PatentType.Application = 'Application';
Srims.patents.PatentType.Design = 'Design';

Srims.patents.PatentTypeRender = function(value, metadata) {
    switch (value) {
        case 'Invention':
            return '发明专利';
        case 'Application':
            return '实用新型';
        case 'Design':
            return '外观设计';
        default:
            return '未知类型';
    }
}


Srims.patents.PatentTypeStore = [['Invention', '发明专利'], ['Application', '实用新型'], ['Design', '外观设计'], ['Unknown', '未知类型']];




